src/command_modules/azure-cli-role/tests/test_role.py

```diff
@@ -8,73 +8,112 @@
+    def test_create_for_rbac_with_secret(self, resource_group):
 
-    def test_create_for_rbac_with_cert(self):
-        if self.playback:
-            return  # live-only test, will enable playback once resolve #635
-        self.execute()
+        subscription_id = self.cmd('account list --query "[?isDefault].id"').get_output_in_json()
+        scope = '/subscriptions/{}'.format(subscription_id[0])
+        sp_name = 'http://{}'.format(resource_group)
 
-    def body(self):
-        subscription_id = self.cmd('account list --query "[?isDefault].id" -o tsv')
-        scope = '/subscriptions/{}'.format(subscription_id)
+        self.cmd('ad sp create-for-rbac -n {0} --scopes {1} {1}/resourceGroups/{2}'.format(sp_name, scope, resource_group), checks=[
+            JMESPathCheckV2('name', sp_name)
+        ])
+        self.cmd('role assignment list --assignee {} --scope {}'.format(sp_name, scope),
+                 checks=[JMESPathCheckV2("length([])", 1)])
+        self.cmd('role assignment list --assignee {} -g {}'.format(sp_name, resource_group),
+                 checks=[JMESPathCheckV2("length([])", 1)])
+        self.cmd('role assignment delete --assignee {} -g {}'.format(sp_name, resource_group),
+                 checks=NoneCheck())
+        self.cmd('role assignment delete --assignee {}'.format(sp_name), checks=NoneCheck())
+        self.cmd('ad app delete --id {}'.format(sp_name))
 
-        result = self.cmd(
-            'ad sp create-for-rbac -n {0} --scopes {1} {1}/resourceGroups/{2} --create-cert'.format(
-                self.sp_name, scope, self.resource_group))
-        self.assertEqual(self.sp_name, result['name'])
+
+class RbacSPCertScenarioTest(LiveTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_create_rbac_sp_with_cert')
+    def test_create_for_rbac_with_cert(self, resource_group):
+
+        subscription_id = self.cmd('account list --query "[?isDefault].id"').get_output_in_json()
+        scope = '/subscriptions/{}'.format(subscription_id[0])
+        sp_name = 'http://' + resource_group
+
+        result = self.cmd('ad sp create-for-rbac -n {0} --scopes {1} {1}/resourceGroups/{2} --create-cert'.format(sp_name, scope, resource_group)).get_output_in_json()
+        self.assertEqual(sp_name, result['name'])
         self.assertTrue(result['fileWithCertAndPrivateKey'].endswith('.pem'))
         os.remove(result['fileWithCertAndPrivateKey'])
-        result = self.cmd('ad sp reset-credentials -n {0} --create-cert'.format(self.sp_name))
-        self.assertEqual(self.sp_name, result['name'])
+        result = self.cmd('ad sp reset-credentials -n {0} --create-cert'.format(sp_name)).get_output_in_json()
+        self.assertEqual(sp_name, result['name'])
         self.assertTrue(result['fileWithCertAndPrivateKey'].endswith('.pem'))
         os.remove(result['fileWithCertAndPrivateKey'])
+        self.cmd('ad app delete --id {}'.format(sp_name), checks=NoneCheck())
```

Comment:
Yes, the teardown method won't work well here because the app IDs are random.

Instead what I did is wrap these commands in try/finally blocks so the app gets deleted if anything after it is created fails.